gradle-idea-configuration/src/main/groovy/com/palantir/gradle/ideaconfiguration/PluginDependency.java

```diff
@@ -62,25 +69,4 @@ public static int compareVersions(String v1, String v2) {
                 .findFirst()
                 .orElse(0);
     }
-
-    @Override
-    public String toString() {
-        return minVersion != null ? name + " (" + minVersion + ")" : name;
-    }
-
-    @Override
-    public boolean equals(Object object) {
-        if (this == object) {
-            return true;
-        }
-        if (!(object instanceof PluginDependency that)) {
-            return false;
-        }
-        return Objects.equals(name, that.name) && Objects.equals(minVersion, that.minVersion);
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(name, minVersion);
-    }
```

Comment:
Generally if you ever find yourself writing equals and hashCode something has gone wrong. The Gradle [managed types](https://github.com/palantir/gradle-guide/blob/develop/guide/managed-types-and-properties.md) will handle this for you (or immutables).